test/source/tests/compose.ts

```diff
@@ -2121,6 +2165,14 @@ export const defineComposeTests = (testVariant: TestVariant, testWithBrowser: Te
       'auto-refresh expired key if newer version of the same key available',
       testWithBrowser('ci.tests.gmail', async (t, browser) => {
         // add an expired key manually
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Is the non-null assertion needed here? Maybe we can modify the style to not use nulls and use undefined instead, then you won't need this line?

Reply:
`mockApi` is defined as optional here because we are extending `ExecutionContext`.
However, we should force unwrap `mockApi` because we know it won't be null as we set `mockApi` value in `testWithBrowser` function.
Therefore I think above eslint disable rule is required.

https://github.com/FlowCrypt/flowcrypt-browser/blob/aaabc42912f751a70a1f6537725571c13a465b34/test/source/tests/tooling/index.ts#L16

Reply:
Alternatively, how about we just disable `no-non-null-assertion` for whole test folder?

Reply:
It's a detail, in either case. Leave as is is also fine with me, just seems inconvenient to always have to worry about this.

Reply:
I was thinking about possible improvement for this one too, probably we can move our properties from [AvaContext](https://github.com/FlowCrypt/flowcrypt-browser/blob/d85f2eac158df355a174e06d0689bf059b6fffc7/test/source/tests/tooling/index.ts#L8) to separate interface, like this:
```ts
interface AvaExecutionContext {
  retry?: true;
  attemptNumber?: number;
  totalAttempts?: number;
  attemptText?: string;
  extensionDir: string;
  urls: TestUrls;
  mockApi: Api<{ query: { [k: string]: string }; body?: unknown }, unknown>;
}

export type AvaContext = ExecutionContext<AvaExecutionContext>

// then it can be used in tests like this
async (t: AvaContext) => {
   t.context.mockApi.attesterConfig = { ... }
}
```

But need to check if it works

Reply:
> Implemented this solution and pushed.
> Please check. Will revert changes if you are not okay

Let's revert it back to optional `mockApi` property, as it's not good practice to use `@ts-ignore`, can lead to other issues in the future. Later we can come back to this issue and try to find another solution.

Reply:
Ok, how about we disable `no-non-null-assertion` rule? Adding `// eslint-disable-next-line @typescript-eslint/no-non-null-assertion` to every `t.mockApi.attesterConfig = ` doesn't seem like good idea too.